site/zenodo_rdm/moderation/percolator.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2017-2024 CERN.
+# Copyright (C) 2022 Graz University of Technology.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Percolator."""
+
+
+from flask import current_app
+from invenio_search import current_search_client
+from invenio_search.utils import build_alias_name, build_index_name
+
+
+def create_percolator_index(index_name, record_cls):
+    """Create mappings with the percolator field for moderation queries.
+
+    This function creates a new Elasticsearch index for percolator queries by copying
+    the settings and mappings from an existing record index and adding specific
+    percolator mappings.
+    """
+    # Retrieve the alias name for the record index
+
+    record_index = build_alias_name(record_cls.index._name)
+    combined_index_name = f"{index_name}-{record_cls.index._name}"
+    # Build the name for the new percolator index, using a prefix and the record's index name
+    percolator_index = build_index_name(combined_index_name, app=current_app)
+
+    # Get the current mapping for the record index to copy its structure
+    record_mapping = current_search_client.indices.get_mapping(index=record_index)
+    # Extract the mappings from the record index and store in `percolator_mappings`
+    percolator_mappings = list(record_mapping.values())[0]["mappings"]
```

Comment:
```suggestion
    assert len(record_mapping) == 1
    percolator_mappings = list(record_mapping.values())[0]["mappings"]
```

minor: just to make sure we're picking the right mapping